contracts/Removal.sol

```diff
@@ -393,15 +353,21 @@ contract Removal is
   }
 
   function _createRemovalDataBatch(
-    uint256[] calldata removalIds,
+    UnpackedRemovalIdV0[] calldata removals,
```

Comment:
potentially concerned about this approach because unless we're passing the ids to `mintBatch`, we don't necessarily know what those ids are going to be, and we can't reliably stamp them onto the removal entities that we need to be keeping track of off-chain.  in writing the first version of the admin minting page, there wasn't a reliable way to depend on events to unequivocally link a minted ID to the removal for which it was being minted.  of course we can generate a removal's id using the `RemovalIdLib`, but unless we then also explicitly pass that id to the mint function, there's still a disconnect that could lead to something being corrupted

Reply:
there emitted in the `TransferBatch` event, no?

Reply:
I honestly hated encoding them off chain as it makes any validation on-chain basically unreliable. Strongly in favor of keeping it this way, but totally onboard to add whatever needed to do what you're asking for assuming the event that's emitted isn't enough

Reply:
(you get zero-cost intrinsic validation by using a struct as the param as you can't pass invalid types, btw)

Reply:
if I recall, the issue is that yes they are emitted in the TransferBatch event but if you have multiple successive mintings happening to the same supplier (and I think even just in a single mint with multiple removals) you just don't have a reliable way to create a perfect 1:1 linkage between the given removal in our database and the exact on chain id it got.  i think in the case that there's only one minting event for the supplier you can rely on the order of the argument in the array to map 1:1 your batch and their ids (though even this feels a little brittle) but the bigger deal is whether you have multiple mints for a supplier

Reply:
it's hard to think about what additional event information could be emitted that would solve for this problem unless some other uniquely identifying off-chain id was connected to the entity and then also passed in as part of `mintBatch` and used as a sort of identifying tag to match up with the official removal id.  like the issue here isn't one of implementation but rather of information actually being unique or not.  I have a feeling that if we go with this approach we are going to end up generating the ids off-chain too with a call to `createRemovalId` and then just hoping that everything was done correctly.

Reply:
We could just return the ids from the mint function. Either way we have to track some data here if we cant just get it on chain. We could also just expose a function that let us paginate through existing IDs. because they definitely do exist on-chain

Reply:
if we need to store a mapping in a db somewhere, seems reasonable to just use the return value as the thing we store

Reply:
transactions don't have return values though... I think that's the issue and why we have to rely on events

Reply:
exposing a function that allows us to paginate through existing ids is something I've thought we might want to have anyway.... but I don't think it helps us with this problem. the ID is the unique identifier of a removal.  unless we say off-chain that THIS removal has THIS ID, and then use that exact piece of data to mint it, there's just no other uniquely identifying way that we can keep the two things completely in sync.  I'm having a lot of opinions about this because this is literally a problem that we have grappled with before and landed in a place where passing the ids made the most sense.  I'm worried about the headache it's going to cause us when we try to finish the app portion of this

Reply:
so i'm up for doing it differently but I just feel like I want to have a clear plan that we know will work before making this commitment and I can't put it all together yet